library/core/src/ffi/c_str.rs

```diff
@@ -295,11 +295,12 @@ impl CStr {
         }
     }
 
-    /// Creates a C string wrapper from a byte slice.
+    /// Creates a C string wrapper from a byte slice that includes one or more
+    /// nul bytes.
```

Comment:
Strictly speaking, it is valid to call this on a byte slice with zero `\0` present, because the slice describes a safe bound to search in. That makes this *technically* incorrect.

In addition, this first line is shown first in many inline editor hints or row-by-row formatted lists and accordingly should be kept ultra-short, so I hesitate at making it so much longer. It's fine if you can't find a way to express that in fewer characters, though.